biothings/hub/__init__.py

```diff
@@ -1506,10 +1519,81 @@ def configure_api_endpoints(self):
             self.api_endpoints.pop("standalone")
         if "upgrade" in self.commands:
             self.api_endpoints["code/upgrade"] = EndpointDefinition(name="upgrade", method="put")
+        if "quick_index" in self.extra_commands:
+            self.api_endpoints["quick_index"] = EndpointDefinition(name="quick_index", method="post", force_bodyargs=True)
 
     def export_command_documents(self, filepath):
         generate_command_documentations(filepath, self.commands)
 
+    def quick_index(
+        self,
+        datasource_name,
+        doc_type,
+        indexer_env,
+        index_name=None,
+        **kwargs,
+    ):
+        """
+        Intention for datasource developers to quickly create an index to test their datasources.
+        Automatically create temporary build config, build collection
+        Then call the index method with the temporary build collection's name
+        """
+
+        random_string = f"{get_timestamp()}_{get_random_string()}"
+        # generate random build_configuration name
+        build_configuration_name = f"{datasource_name}_configuration_{random_string}"
+        # generate random build name
+        build_name = f"{datasource_name}_{random_string}"
+        # # generate index_name if needed
+        if not index_name:
+            index_name = build_name
+        index_name = index_name.lower()
+
+        async def do():
+            extra_index_settings = kwargs.pop("extra_index_settings", '{}')
+            extra_index_settings = json.loads(extra_index_settings)
+            extra_index_settings["num_shards"] = int(kwargs.pop("num_shards", 1))
+            extra_index_settings["num_replicas"] = int(kwargs.pop("num_replicas", 1))
```

Comment:
set default to 0, consistent with:

https://github.com/biothings/biothings.api/blob/1ba90f9a501f3810f29470a370a1413124e17960/biothings/hub/dataindex/indexer.py#L130-L136

Reply:
I made a quick commit.

Reply:
Thanks